run-tests.php

```diff
@@ -2723,8 +2723,12 @@ function run_test(string $php, $file, array $env): string
     if (!$passed || $leaked) {
         // write .sh
         if (strpos($log_format, 'S') !== false) {
-            $env_lines = [];
+            $env_lines = ['unset $(env | cut -d= -f1)'];
             foreach ($env as $env_var => $env_val) {
+                if (strval($env_val) === '') {
+                    // proc_open does not pass empty env vars
+                    continue;
+                }
```

Comment:
`proc_open` does not pass empty env vars, so if we export empty env vars here, the test will see these vars when running with the `.sh` file, but not when running it via `run-tests.php`.

Reply:
Ah, that's the spot, here is how I did it working, should have probably posted that earlier, sigh:

```php
$descriptor_spec = [];
$env   = ['TRUTH='];
$pipes = [];
$proc  = proc_open(['env'], $descriptor_spec, $pipes, null, $env);
$exit  = proc_close($proc);
```
output:

```
TRUTH=
```

Thanks for double checking @arnaud-lb, this clarifies the comment in the run-tests.php script for me, it depends on passing the $env_vars as list or associative.